File: app/src/main/java/org/descartae/android/view/fragments/empty/EmptyLocationPermissionFragment.java
package org.descartae.android.view.fragments.empty;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import org.descartae.android.R;
import org.descartae.android.interfaces.RequestPermissionView;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class EmptyLocationPermissionFragment extends Fragment {

    private RequestPermissionView mListener;

    public EmptyLocationPermissionFragment() {
    }

    public static EmptyLocationPermissionFragment newInstance() {
        EmptyLocationPermissionFragment fragment = new EmptyLocationPermissionFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments() != null) {

        }
    }

    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_permission_empty, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof RequestPermissionView) {
            mListener = (RequestPermissionView) context;
        } else {
            throw new RuntimeException(context.toString() + " must implement RequestPermissionView");
        }
    }

    @OnClick(R.id.action_allow)
    public void actionAllow() {
        mListener.onAcceptPermission();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }
}

File: app/src/main/java/org/descartae/android/view/fragments/empty/EmptyLocationPermissionFragment.kt
package org.descartae.android.view.fragments.empty

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_permission_empty.*
import org.descartae.android.R
import org.descartae.android.interfaces.RequestPermissionView


class EmptyLocationPermissionFragment : Fragment() {

    private var mListener: RequestPermissionView? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        action_allow.setOnClickListener { mListener?.onAcceptPermission() }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_permission_empty, container, false)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is RequestPermissionView) {
            mListener = context
        } else {
            throw RuntimeException(context!!.toString() + " must implement RequestPermissionView")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    companion object {

        fun newInstance(): EmptyLocationPermissionFragment {
            val fragment = EmptyLocationPermissionFragment()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }
}
